pkg/pbq/store/memory/stores.go

```diff
@@ -60,3 +71,15 @@ func (ms *memoryStores) DiscoverPartitions(ctx context.Context) ([]partition.ID,
 	}
 	return []partition.ID{}, nil
 }
+
+func (ms *memoryStores) DeleteStore(partitionID partition.ID) error {
+	memStore, ok := ms.partitions[partitionID]
+	if !ok {
+		return errors.New("store not found")
+	}
+
+	memStore.storage = nil
+	memStore.writePos = -1
+	delete(ms.partitions, partitionID)
```

Comment:
Does it need a lock?

Reply:
No, I don't think so, because the store will be associated with only one partition

Reply:
oh you mean the map, yeah it needs a lock